default/methods/task/task/task_user_add.py

```diff
@@ -17,19 +18,26 @@ def api_task_user_add(project_string_id, task_id):
             }}
         ]
 
+        result = {}
+
         log, input, untrusted_input = regular_input.master(request = request,
                                                            spec_list = spec_list)
         if len(log["error"].keys()) >= 1:
             return jsonify(log = log), 400
+        
+        for user in input['user_id']:
+            users_already_assigned = session.query(TaskUser).filter(TaskUser.task_id == task_id).filter(TaskUser.relation ==  input['relation']).filter(TaskUser.user_id == user).count()
```

Comment:
Hey Vitalii!

Thanks for adding the support for lists here. As a small convention on the backend, we always try to move any logic to the core logic layer instead of the API function.

So in this example, what I would do is just move the user forloop inside the `api_task_user_add_core` function. That way we can leave in the API just the permissions and the payload type checking. 

In general we always do it in the following way:

1. They api function ( in this case `api_task_user_add`) has the responsability of checking permissions (with the decorators or any other specifics if necessary),  creating the DB session, checking the payload types are correct, and then calling the core function that will process the actual business logic.
2. The core function (in this case `api_task_user_add_core`) has the responsabiliy of executing the business logic and returning the serialized result so that the API function can return it to the client.

Reply:
Got it